libc/include/__llvm-libc-common.h

```diff
@@ -47,6 +47,9 @@
 #define __NOEXCEPT throw()
 #endif
 
+#undef LLVM_LIBC_CAST
+#define LLVM_LIBC_CAST(cast, type, value) (cast<type>(value))
```

Comment:
This merits a comment about its intended uses.
No macro should be defined in this file unless it's in the implementation namespace.
Sticking to the `__` prefix seems like the best convention, with the exception of the C11 keywords that are in the `_[A-Z]` part of the implementation namespace.